usr/lib/pkcs11/icsf_stdll/icsf_specific.c

```diff
@@ -595,20 +595,10 @@ CK_RV reset_token_data(STDLL_TokData_t *tokdata, CK_SLOT_ID slot_id,
 
 	/* Reset token data and keep token name */
 	slot_data[slot_id]->initialized = 0;
-	init_token_data(tokdata, slot_id);
+    load_token_data(tokdata, slot_id);
 	init_slotInfo(&tokdata->slot_info);
 	tokdata->nv_token_data->token_info.flags |= CKF_TOKEN_INITIALIZED;
 
-	/* Reset SO pin to default and user pin to invalid */
```

Comment:
I don't understand this part of the patch.

Reply:
ICSF token, specifically (the other tokens don't do that), was resetting the SO PIN to the default "87654321" and the User PIN to an invalid PIN. 
According to the PKCS#11 this shouldn't be done. 
If you are reinitializing a token, the SO PIN that you will use during this reinitialization must be the PIN that was set before the reinitialization. 
That's why I removed this piece of code as this is not compliant to the standard.

Reply:
Thanks or the explanation.